src/reducers/editThought.ts

```diff
@@ -32,6 +45,36 @@ const editThought = (
 
   const editedThought = getThoughtById(state, editedThoughtId)
 
+  const editedThoughtValue = editedThought.value
+
+  const parentThoughtIdsForCollidedThoughts = thoughtCollision?.contexts.map(
+    thoughtId => parentOfThought(state, thoughtId)?.id,
+  )
```

Comment:
You should not be using `thoughtCollision` at all. It checks if the new value exists anywhere else in the thoughtspace. This functionality is only concerned with sibling thoughts, and should not be iterating over every other thought in the thoughtspace with the same value.

You can use `Thought.childrenMap` for O(1) lookup of meta attributes, as they are keyed by value:

```ts
const styleThoughtId = thought.childrenMap['=style']
```

Reply:
Even better, use https://github.com/cybersemics/em/blob/dev/src/selectors/findDescendant.ts. That way the implementation details are hidden. It is optimized for attribute children.